### ISSUE TITLE
TikZjax figure support.

### ISSUE THREAD

**Issue by ROCKTAKEY** [kaushalmodi/ox-hugo#390]
Hi! Thanks for your great package.

I would like to use [TikZJax](https://tikzjax.com/), which draws svg image from source, on ox-hugo.
I'm glad if I can write on org-mode like:

```org-mode
* Some head
  #+caption: Circle
  #+label: circle
  #+begin_export html
  <script type="text/tikz">
   \begin{tikzpicture}
    \draw (0,0) circle (1in);
   \end{tikzpicture}
  </script>
  #+end_export
```

Currently, the caption and label are ignored, so the TikZJax figure cannot be referred or titled. Is it possible to support this?

Thanks!

**Comment by kaushalmodi** [comment]
Hello, 

Thank you!

Unfortunately, I am not able to spend time on adding new features (because of work load). I can suggest that you looking into submitting a PR for this. What you want can probably work if you add support using "Org Special Blocks". As an example, see how I support `#+begin_description .. #+end_description` [here](https://github.com/kaushalmodi/ox-hugo/blob/75b849e9561c4a6022babf6eaf0e037310ded7c1/ox-hugo.el#L2746). In the same way, you can add support for `#+begin_tikz .. #+end_tikz`.

**Comment by kaushalmodi** [comment]
> Currently, the caption and label are ignored, so the TikZJax figure cannot be referred or titled. Is it possible to support this?

I don't think that's possible to do easily because `ox.el` (the core library of all Org exporters) doesn't support labeling and captioning arbitrary elements. You can try exporting the same to HTML using ox-html and you will find the same.

Though, I do have a workaround, using the Org supported [`<<target>>` type internal links](https://orgmode.org/manual/Internal-Links.html) and HTML `figcaption`:

```org
<<circle>>
#+begin_export html
<figure>
  <script type="text/tikz">
    \begin{tikzpicture}
    \draw (0,0) circle (1in);
    \end{tikzpicture}
  </script>
<figcaption>Circle</figcaption>
#+end_export

Here's a link to the [[circle][Circle]].
```

This works.

**Comment by kaushalmodi** [comment]
After https://github.com/kaushalmodi/ox-hugo/pull/510, this is now possible:

```org
<<circle>>
#+attr_html: :caption Picture of a circle
#+begin_tikzjax
\draw (0,0) circle (1in);
#+end_tikzjax

Here's a link to the [[circle][Circle]].
```

**Comment by ROCKTAKEY** [comment]
Sorry for late reply.
It seems so good! Thanks a lot!
